ownCloud/Settings/Passcode/PasscodeViewController.swift

```diff
@@ -199,36 +199,44 @@ class PasscodeViewController: UIViewController, Themeable {
 
 	// MARK: - Actions
 	@IBAction func appendDigit(_ sender: UIButton) {
-		if !keypadButtonsEnabled || keypadButtonsHidden {
-			return
-		}
-
-		if let currentPasscode = passcode {
-			// Enforce length limit
-			if currentPasscode.count < passcodeLength {
-				self.passcode = currentPasscode + String(sender.tag)
-			}
-		} else {
-			self.passcode = String(sender.tag)
-		}
-
-		// Check if passcode is complete
-		if let passcode = passcode {
-			if passcode.count == passcodeLength {
-				// Delay to give feedback to user after the last digit was added
-				OnMainThread(after: 0.1) {
-					self.completionHandler?(self, passcode)
-				}
-			}
-		}
+        appendDigit(digit: String(sender.tag))
 	}
+    
+    private func appendDigit(digit: String) {
+        if !keypadButtonsEnabled || keypadButtonsHidden {
+            return
+        }
+        
+        if let currentPasscode = passcode {
+            // Enforce length limit
+            if currentPasscode.count < passcodeLength {
+                self.passcode = currentPasscode + String(digit)
```

Comment:
@hosy digit is already String. String(String) doesn't make sense

Reply:
Makes sense. I put the code in a new method and did not make changes to the existing code